docs/modules/ROOT/pages/subscriptions/mechanisms.adoc

```diff
@@ -0,0 +1,146 @@
+[[subscription-mechanisms]]
+= Subscription Mechanisms
+
+There are different ways in how a GraphQL subscription may be set along with `@neo4j/graphql` server.
+
+
+== Default
+The default behaviour is automatically set if the `subscriptions` feature is set to true, as described in xref::subscriptions/getting-started.adoc[Getting Started]:
+
+```javascript
+new Neo4jGraphQL({
+    typeDefs,
+    driver,
+    features: {
+        subscriptions: true
+    },
+});
+```
+
+This behaviour enables a simple subscription system that will work on a single instance, which is ideal for development, testing, and servers that do not require horizontal scaling.
+
+[[amqp]]
+== AMQP
+Using subscriptions on a server with multiple instances can be tricky, as described in xref::subscriptions/scaling.adoc[Scaling]. For this, the recommended approach is to use a PubSub system, which can be achieved with an AMQP broker such as link:https://www.rabbitmq.com/[RabbitMQ]. This is supported through the link:https://www.npmjs.com/package/@neo4j/graphql-plugin-subscriptions-amqp[@neo4j/graphql-plugin-subscriptions-amqp] package.
+
+The `@neo4j/graphql-plugin-subscriptions-amqp` plugin connects to message brokers through `AMQP 0-9-1` protocol to distribute subscription events across all server instances. 
+
+Some brokers supporting this protocol are:
+
+* link:https://www.rabbitmq.com/[RabbitMQ]
+* link:https://qpid.apache.org/[Apache Qpid]
+* link:https://activemq.apache.org/[Apache ActiveMQ]
+
+It can be installed with npm:
+
+```sh
+npm install @neo4j/graphql-plugin-subscriptions-amqp
+```
+
+NOTE: AMQP 1.0 is **not** supported by this plugin.
+
+=== Usage
+
+The AMQP plugin should be instanced and passed to the `subscription` field in features, this will automatically enable the subscriptions with the AMQP broker as a message queue:
+
+```javascript
+const { Neo4jGraphQLSubscriptionsAMQPPlugin } = require("@neo4j/graphql-plugin-subscriptions-amqp");
+
+const amqpSubscription = new Neo4jGraphQLSubscriptionsAMQPPlugin({
+    connection: {
+        hostname: "localhost",
+        username: "guest",
+        password: "guest",
+    }
+});
+
+const neoSchema = new Neo4jGraphQL({
+    typeDefs,
+    driver,
+    features: {
+        subscriptions: amqpSubscription,
+    },
+});
+```
+
+=== API
+The following options can be passed to the constructor:
+
+* **connection**: An AMQP uri as a string or a configuration object:
+** **hostname**: Hostname to be used, defaults to `localhost`.
+** **username**: defaults to `guest`.
+** **password**: defaults to `guest`.
+** **port**: Port of the AMQP broker, defaults to `5672`.
+* **exchange**: The exchange to be used in the broker. Defaults to `neo4j.graphql.subscriptions.fx`.
+* **version**: The AMQP version to be used. Currently only `0-9-1` is supported.
+
+Additionally, any option supported by link:https://www.npmjs.com/package/amqplib[amqplib] can be passed to `connection`.
```

Comment:
Shouldn't this line be under "Methods"?

Reply:
Mmm, not really. This refers to the `connection` property described above, maybe we could move it there to avoid confusion?